autotest/gdrivers/heif.py

```diff
@@ -158,3 +161,136 @@ def test_heif_subdatasets():
         assert gdal.Open("HEIF:1") is None
     with pytest.raises(Exception):
         assert gdal.Open("HEIF:1:") is None
+
+
+def make_data():
+    ds = gdal.GetDriverByName("MEM").Create("", 300, 200, 3, gdal.GDT_Byte)
+
+    ds.GetRasterBand(1).SetRasterColorInterpretation(gdal.GCI_RedBand)
+    ds.GetRasterBand(2).SetRasterColorInterpretation(gdal.GCI_GreenBand)
+    ds.GetRasterBand(3).SetRasterColorInterpretation(gdal.GCI_BlueBand)
+
+    red_green_blue = (
+        ([0xFF] * 100 + [0x00] * 200)
+        + ([0x00] * 100 + [0xFF] * 100 + [0x00] * 100)
+        + ([0x00] * 200 + [0xFF] * 100)
+    )
+    rgb_bytes = array.array("B", red_green_blue).tobytes()
+    for line in range(100):
+        ds.WriteRaster(
+            0, line, 300, 1, rgb_bytes, buf_type=gdal.GDT_Byte, band_list=[1, 2, 3]
+        )
+    black_white = ([0xFF] * 150 + [0x00] * 150) * 3
+    black_white_bytes = array.array("B", black_white).tobytes()
+    for line in range(100):
+        ds.WriteRaster(
+            0,
+            100 + line,
+            300,
+            1,
+            black_white_bytes,
+            buf_type=gdal.GDT_Byte,
+            band_list=[1, 2, 3],
+        )
+
+    assert ds.FlushCache() == gdal.CE_None
+    return ds
+
+
+def make_data_with_alpha():
+    ds = gdal.GetDriverByName("MEM").Create("", 300, 200, 4, gdal.GDT_Byte)
+
+    ds.GetRasterBand(1).SetRasterColorInterpretation(gdal.GCI_RedBand)
+    ds.GetRasterBand(2).SetRasterColorInterpretation(gdal.GCI_GreenBand)
+    ds.GetRasterBand(3).SetRasterColorInterpretation(gdal.GCI_BlueBand)
+    ds.GetRasterBand(4).SetRasterColorInterpretation(gdal.GCI_AlphaBand)
+
+    red_green_blue_alpha = (
+        ([0xFF] * 100 + [0x00] * 200)
+        + ([0x00] * 100 + [0xFF] * 100 + [0x00] * 100)
+        + ([0x00] * 200 + [0xFF] * 100)
+        + ([0x7F] * 150 + [0xFF] * 150)
+    )
+    rgba_bytes = array.array("B", red_green_blue_alpha).tobytes()
+    for line in range(100):
+        ds.WriteRaster(
+            0, line, 300, 1, rgba_bytes, buf_type=gdal.GDT_Byte, band_list=[1, 2, 3, 4]
+        )
+    black_white = ([0xFF] * 150 + [0x00] * 150) * 4
+    black_white_bytes = array.array("B", black_white).tobytes()
+    for line in range(100):
+        ds.WriteRaster(
+            0,
+            100 + line,
+            300,
+            1,
+            black_white_bytes,
+            buf_type=gdal.GDT_Byte,
+            band_list=[1, 2, 3, 4],
+        )
+
+    assert ds.FlushCache() == gdal.CE_None
+    return ds
+
+
+heif_codecs = ["HEIF", "JPEG", "JPEG2000"]
+
+
+@pytest.mark.parametrize("codec", heif_codecs)
+def test_heif_create_copy(codec):
+    tempfile = "tmp/test_heif_create_copy_" + codec + ".hif"
+    try:
+        os.remove(tempfile)
+    except OSError:
+        pass
```

Comment:
```suggestion
def test_heif_create_copy(tmp_path, codec):
    tempfile = str(tmp_path / "test_heif_create_copy_" + codec + ".hif")
```

and same in below similar occurences